From 4bc3859fd2c1b6243f69e7e78237c0b6c1ac2ca3 Mon Sep 17 00:00:00 2001
From: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date: Wed, 11 May 2016 14:03:04 -0500
Subject: [PATCH] drm/i915: fix LPE audio issues when resolution changes

Make sure N/CTS parameters are updated on each resolution
change, not just at boot

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index d0e7af2..6540046 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -7969,7 +7969,7 @@ static int i9xx_crtc_compute_clock(struct intel_crtc *crtc,
 
 	/* Added for HDMI Audio */
 	if ((IS_CHERRYVIEW(dev)) || (IS_VALLEYVIEW(dev))) {
-		if(intel_pipe_has_type(crtc, INTEL_OUTPUT_HDMI)) {
+		if (intel_pipe_will_have_type(crtc_state, INTEL_OUTPUT_HDMI)) {
 			dev_priv->tmds_clock_speed = crtc_state->port_clock;
 
 			mid_hdmi_audio_signal_event(dev_priv->dev,
